hicommonwealth/commonwealth
# Add external URL linking card and modal

## Link to Issue
Closes: #4829 

## Description of Changes
- Adds support for `LinkSource.Web` type links via a link card on view thread page
- Includes a modal for creating and deleting new URL links
- Includes ability to title links 

<img width="677" alt="image" src="https://github.com/hicommonwealth/commonwealth/assets/36428666/1667f890-4f8b-405e-9151-9850dc2e65de">

<img width="622" alt="image" src="https://github.com/hicommonwealth/commonwealth/assets/36428666/e01571b2-65f5-4d48-956f-0c0a5d59bab8">
<img width="335" alt="image" src="https://github.com/hicommonwealth/commonwealth/assets/36428666/7ef267bb-556e-4829-9239-9a3ee1a5f8d6">


## Test Plan
- Go to a thread where you are an author or admin
- Click on 'manage Links' in the new card
- Add a link of your choice and select +, add as many links as desired
- Ensure links populate and work on thread page

- Click manage links 
- De-select a link you wish to delete and save changes
- ensure link is deleted


## Other Considerations

- See conversation in ticket. There is an additional(potentially) user story to tag 'proposal external links' which is still in discussion and can be planned to be included in this ticket(probably a 2 point addition)
- One work around idea was that users can title links anything with these changes so the user story could be achieved in the meantime without tags
